pkg/crc/preflight/preflight_linux.go

```diff
@@ -106,18 +125,72 @@ var libvirtPreflightChecks = [...]Check{
 		cleanupDescription: "Removing /etc/NetworkManager/dnsmasq.d/crc.conf file",
 		cleanup:            removeCrcDnsmasqConfigFile,
 	},
-	{
-		cleanupDescription: "Removing the crc VM if exists",
-		cleanup:            removeCrcVM,
-		flags:              CleanUpOnly,
-	},
 }
 
-func getPreflightChecks(experimentalFeatures bool) []Check {
-	checks := []Check{}
+var vsockPreflightChecks = Check{
+	configKeySuffix:  "check-vsock",
+	checkDescription: "Checking if vsock is correctly configured",
+	check:            checkVsock,
+	fixDescription:   "Checking if vsock is correctly configured",
+	fix:              fixVsock,
+}
+
+func checkVsock() error {
+	info, err := os.Stat("/dev/vsock")
+	if err != nil {
+		return err
+	}
+	if stat, ok := info.Sys().(*syscall.Stat_t); ok {
+		group, err := user.LookupGroupId(fmt.Sprint(stat.Gid))
+		if err != nil {
+			return err
+		}
+		if group.Name != "libvirt" {
+			return errors.New("/dev/vsock is not is the right group")
+		}
+	} else {
+		return errors.New("cannot cast info")
+	}
+	if info.Mode()&0060 == 0 {
+		return errors.New("/dev/vsock doesn't have the right permissions")
+	}
+	return nil
+}
+
+func fixVsock() error {
+	_, _, err := crcos.RunWithPrivilege("modprobe vhost_vsock", "modprobe", "vhost_vsock")
```

Comment:
This won't persist on reboot, which means `crc setup` will need to be run again after rebooting.

Reply:
very good point! Should I add a file in /etc/modprobe.d ?